services/dashboard/search.rb

```diff
@@ -0,0 +1,29 @@
+class DashboardService
+  class Search < ApplicationService
+    def initialize(params)
+      @departure_id   = params[:departure_id]
+      @arrival_id     = params[:arrival_id]
+    end
+
+    def call
+      found_routes = Route.where(
+        start_id:       @departure_id,
+        destination_id: @arrival_id
+      )
+
+      routes = found_routes.map { |route| present(route) }
+    end
+
+    private
+
+    def present(route)
+      {
+        route_id:           route.id,
+        departure_station:  City.find_by_id(@departure_id)&.title,
```

Comment:
Use `find_by(id:...)` and extract finders that operate with DB to a separate methods with memoization